### DISCUSSION TITLE
`onLinkClick` in Pivot Not Functioning Correctly When Inside `fluentPage`

### DISCUSSION THREAD

**Discussion by justacec** [Appsilon/shiny.fluent#167] [Q&A] [answered]
### Steps to reproduce

1. Enter the following code into an R session:

```
library(shiny)
library(shiny.fluent)

shinyApp(
  ui = fluentPage(
    textOutput("text"),
    Pivot(
      onLinkClick = JS('alert("Here")'),
      PivotItem(id = "tab1", headerText = "Apples"),
      PivotItem(id = "tab2", headerText = "Bananas")
    )
  ),
  server = function(input, output) {
  }
)
```

### Bug description

The JS alert code is fired when the page is rendered and is not fired as expected when the user clicks on the `PivotItem` links.

### Expected behavior

The JS code would fire each time the user changes to a different `PivotItem`

### Comments

If the call to `fluentPage` is replaced with a `tagList` call, then the JS code executes as expected.

Relevant software versions:
R: 4.2.0
Shiny: 1.7.3
shiny.fluent: 0.3.0

**Comment by jakubsob** [comment, marked as answer]
Hi @justacec,

this is not a bug. `onLinkClick` is a prop that should be a function. Please take a look [at the docs](https://developer.microsoft.com/en-us/fluentui#/controls/web/pivot) of this component.

![image](https://user-images.githubusercontent.com/37193264/223977824-7c1a0363-8edb-4089-8e5e-2b23937eb80b.png)

Please take a look at this example:
```r
library(shiny)
library(shiny.fluent)

shinyApp(
  ui = tagList(
    textOutput("text"),
    Pivot(
      onLinkClick = JS('() => alert("Here")'),
      PivotItem(id = "tab1", headerText = "Apples"),
      PivotItem(id = "tab2", headerText = "Bananas")
    )
  ),
  server = function(input, output) {
  }
)
```
Converting the JS code passed to `onLinkClick` argument to a function should solve this issue. It works with both `fluentPage` and `tagList`.
